src/web/app/src/components/DependenciesTable/Row.tsx

```diff
@@ -0,0 +1,233 @@
+
+import { dependencyDiscoveryUrl } from '../../config';
+
+const useStyles = makeStyles((theme) => ({
+  rowTitle: {
+    fontSize: '14px',
+    color: theme.palette.text.secondary,
+    fontWeight: 700,
+  },
+  rowTitleIcon: {
+    marginRight: '1rem',
+    fontSize: '2rem',
+  },
+  iconRow: {
+    textAlign: 'right !important' as 'right',
+  },
+  dependencyInfo: {
+    display: 'inline-flex',
+    alignItems: 'center',
+    margin: '1rem',
+  },
+  dependencyInfoIssue: {
+    margin: '1rem',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  dependencyInfoIcon: {
+    fontSize: '2rem !important',
+    marginRight: '1.5rem',
+  },
+  issueCard: {
+    display: 'flex',
+    border: '1px',
+    borderColor: theme.palette.text.primary,
+    borderStyle: 'solid',
+    color: theme.palette.text.primary,
+    padding: '1rem',
+    marginTop: '1rem',
+    marginBottom: '1rem',
+  },
+  issueCardLink: {
+    textDecoration: 'none',
+  },
+  collapseBox: {
+    fontSize: '14px',
+    color: theme.palette.text.primary,
+    padding: '2rem',
+    fontFamily: 'Spartan',
+  },
+  issueCardTitle: {
+    fontWeight: 700,
+    fontFamily: 'Spartan',
+  },
+}));
+
+type issueInfo = {
+  htmlUrl: string;
+  title: string;
+  body: string;
+  createdAt: string;
+};
+
+type RowProps = {
+  dependency: string;
+};
+
+export const Row = ({ dependency }: RowProps) => {
+  const classes = useStyles();
+  const [open, setOpen] = useState<boolean>(false);
+  const [url, setUrl] = useState<string>('');
+  const [directory, setDirectory] = useState<string>('');
+  const [license, setLicense] = useState<string>('');
+  const [issues, setIssues] = useState<Array<issueInfo>>([]);
+  const [issueURL, setIssueUrl] = useState<string>('');
+  const [apiLimited, setApiLimited] = useState<boolean>(false);
+
+  // Fetching more info of the dependency
+  const fetchMoreDependencyInfo = async () => {
+    const projectInfo = await fetch(`${dependencyDiscoveryUrl}/projects/${dependency}`);
+    const { gitRepository: { url, directory } = { url: '', directory: '' }, license } =
+      await projectInfo.json();
+    setUrl(url);
+    setIssueUrl(
+      `${url}/issues?q=is%3Aopen+is%3Aissue+label%3A"good+first+issue"%2C"%27hacktoberfest"%2C"help+wanted"+`
+    );
+    setDirectory(directory);
+    setLicense(license);
+  };
+
+  // Fetching opened issue the dependency
+  const fetchDependencyOpenedIssue = async () => {
+    setApiLimited(false);
+    const issueInfo = await fetch(`${dependencyDiscoveryUrl}/github/${dependency}`);
+
+    if (issueInfo.status === 403) {
+      setApiLimited(true);
+      return;
+    }
+    const issuesInfoToJson = await issueInfo.json();
+    setIssues(issuesInfoToJson);
+  };
+
+  // Handle collapse action
+  // Fetch more info only on collapse open
+  const handleCollapseAction = async () => {
+    if (!open) {
+      try {
+        await Promise.all([fetchMoreDependencyInfo(), fetchDependencyOpenedIssue()]);
```

Comment:
I think we need a spinner or something when this starts, and end it in a `.finally()`, because it can take some time before the row opens.